### PR TITLE
fix: reduce proof requests

### PR DESCRIPTION
**Pull request by Lanttcat**
## Description

Closes # (NO_ISSUE)

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/dashboard/src/pages/Personas/hooks/usePersonaProofs.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    })
+
+    const retry = () => {
+        if (state.loading) return
+        fn(false)
+    }
+
+    useEffect(() => Messages.events.ownProofChanged.on(retry), [retry])
+
+    return {
```
Comment (guanbinrui) [inline packages/dashboard/src/pages/Personas/hooks/usePersonaProofs.ts]: Return as `AsyncStateRetry`.

File: packages/mask/src/utils/hooks/usePersonaProofs.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import { EMPTY_LIST } from '@masknet/shared-base'
+import { NextIDProof } from '@masknet/web3-providers'
+import { useEffect } from 'react'
+import { useAsyncFn, useEffectOnce } from 'react-use'
+import { MaskMessages } from '../messages.js'
+
+export function usePersonaProofs(publicKey?: string) {
```
Comment (guanbinrui) [inline packages/mask/src/utils/hooks/usePersonaProofs.ts]: A duplicate hook.

File: packages/web3-providers/src/NextID/proof.ts [hunk 3]
```diff
@@ -107,7 +107,7 @@ export class NextIDProofAPI implements NextIDBaseAPI.Proof {
             urlcat(BASE_URL, '/v1/proof', {
                 platform,
                 identity,
-                page,
+                page: page ?? 1,
```
Comment (guanbinrui) [inline packages/web3-providers/src/NextID/proof.ts]:
```ts
async queryExistedBindingByPlatform(platform: NextIDPlatform, identity: string, page = 1)
```

File: packages/dashboard/src/pages/Personas/components/PersonaCard/Row.tsx [hunk 4]
```diff
@@ -109,7 +109,7 @@ export const PersonaRowCardUI = memo<PersonaRowCardUIProps>((props) => {
     const { onConnect, onDisconnect, onRename, onDeleteBound } = props
     const { value: privateKey } = useExportPrivateKey(identifier)
     const { value: words } = useExportMnemonicWords(identifier)
-    const proof = usePersonaProof(publicKey)
+    const proof = usePersonaProofs(publicKey)
```
Comment (guanbinrui) [inline packages/dashboard/src/pages/Personas/components/PersonaCard/Row.tsx]:
```suggestion
    const proofs = usePersonaProofs(publicKey)
```

File: packages/dashboard/src/pages/Personas/components/PersonaCard/index.tsx [hunk 5]
```diff
@@ -83,7 +83,7 @@ export const PersonaCardUI = memo<PersonaCardUIProps>((props) => {
     const { nickname, active = false, definedSocialNetworks, identifier, profiles, publicKey } = props
     const { onConnect, onDisconnect, onClick } = props
     const { classes, cx } = useStyles()
-    const proof = usePersonaProof(publicKey)
+    const proof = usePersonaProofs(publicKey)
```
Comment (guanbinrui) [inline packages/dashboard/src/pages/Personas/components/PersonaCard/index.tsx]:
```suggestion
    const proofs = usePersonaProofs(publicKey)
```

File: packages/mask/src/plugins/Tips/hooks/useProvedWallets.ts [hunk 6]
```diff
@@ -19,17 +18,15 @@ export function useProvedWallets() {
...
-    }, [currentPersonaIdentifier, personas])
+    const currentPersona = personas?.find(
+        (x) => x.identifier === ECKeyIdentifier.from(currentIdentifier).unwrapOr(head(personas)?.identifier),
+    )
 
-    useEffect(() => MaskMessages.events.ownProofChanged.on(res.retry), [res.retry])
+    const proofs = usePersonaProofs(currentPersona?.identifier.publicKeyAsHex)
 
-    return res
+    return {
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Tips/hooks/useProvedWallets.ts:44]: Return as `AsyncStateRetry`.

File: packages/mask/src/utils/hooks/usePersonaProofs.ts (hunk 2 above)
Comment (Lanttcat) [reply packages/mask/src/utils/hooks/usePersonaProofs.ts]: The hook need use the `MaskMessage`. Not have common package.

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**